src/main/java/org/myrobotlab/service/Solr.java

```diff
@@ -75,6 +80,10 @@ public class Solr extends Service implements DocumentListener, TextListener, Mes
   public String solrHome = "Solr";
   // EmbeddedSolrServer embeddedSolrServer = null;
   transient private EmbeddedSolrServer embeddedSolrServer = null;
+  
+  Collection<SolrInputDocument> documentBatch = Collections.synchronizedCollection(new ArrayList<>());
+  // The batch size of documents to accumulate before flushing the batch to solr.
+  public transient int batchSize = 100;

```

Comment:
Does this have a getter and setter? Might be nice to be able to set this through a message. Side note: might be able to allow setting and getting fields with messages in the future with a bit of tweaking in the method invocation code